src/widget/wwaveformviewer.cpp

```diff
@@ -286,3 +305,8 @@ void WWaveformViewer::unhighlightMark(WaveformMarkPointer pMark) {
 bool WWaveformViewer::isPlaying() const {
     return m_pPlayEnabled->get();
 }
+
+void WWaveformViewer::slotDownbeatUpdated() {
+    m_waveformWidget->getTrackInfo()->getBeats()->setDownbeatStartIndex(
+            m_hoveredBeat.getIndex() % kDefaultBeatsPerMeasure);
+}
```

Comment:
This feels hacky to do this in WWaveformViewer. I propose moving this into WBeatMenu and setting the active Beat from WWaveformViewer, similar to how we set the active Cue for WCueMenuPopup.